.travis.yml

```diff
@@ -13,3 +20,9 @@ env:
 
 script:
    - make "$ITEST_TARGET"
+
+after_script:
+   - if [ -n "${ARTIFACTS_SECRET:-}" ]; then
+         echo 'Build artifacts are available at:';
+         find dist -type f | xargs --replace echo "* https://yelp-travis-artifacts.s3.amazonaws.com/dumb-init/${TRAVIS_BUILD_NUMBER}/${ITEST_TARGET}/{}";
```

Comment:
If you:

```
find ... | xargs -n1 -P0 —replace ...
```

Then it will upload all of the files in parallel.  I do something similar in the following example code https://github.com/samrocketman/jenkins-bootstrap-shared/blob/14b131cb8de289733b4cb3a2c40a4d86992a538a/scripts/uploadRelease.sh#L123-L124

Reply:
This bit is just printing the URLs out at the end, the actual uploading is done via Travis's artifacts addon: https://docs.travis-ci.com/user/uploading-artifacts/

Reply:
I missed that.